src/OAuthAccessToken.php

```diff
@@ -0,0 +1,128 @@
+            }
+            if ('password' === $params['grant_type'] && (!isset($params['username']) || !isset($params['password']))) {
+                throw new \Exception("'password' grant_type required a username and a password");
+            }
+        }
+    }
+
+    public function getConfig()
+    {
+        return [
+            'headers' => [
+                'Authorization' => 'Bearer '.$this->getAccessToken(),
+            ],
+        ];
+    }
+
+    public function getAccessToken($scope = 'base', $noCache = false)
+    {
+        if ($noCache || $this->isCacheExpired()) {
+            $multipartBody = [
+                [
+                    'name' => 'client_id',
+                    'contents' => $this->clientId,
+                ],
+                [
+                    'name' => 'client_secret',
+                    'contents' => $this->clientSecret,
+                ],
+                [
+                    'name' => 'grant_type',
+                    'contents' => isset($this->params['grant_type']) ? $this->params['grant_type'] : 'client_credentials',
+                ],
+                [
+                    'name' => 'scope',
+                    'contents' => 'base',
+                ],
+            ];
+            if (isset($this->params['grant_type']) && $this->params['grant_type'] === 'password') {
+                $multipartBody[] = [
+                    'name' => 'username',
+                    'contents' => $this->params['username'],
+                ];
+                $multipartBody[] = [
+                    'name' => 'password',
+                    'contents' => $this->params['password'],
+                ];
+            }
+            $client = new GuzzleClient(['base_uri' => $this->baseUri, 'http_errors' => false]);
+            $res = $client->request(
+                'POST',
+                '/oauth2/token',
+                [
+                    'multipart' => $multipartBody,
+                ]
+            );
+            if ($res->getStatusCode() > 399) {
+                throw new \Exception(sprintf("Couldnt get a token: (%s):\n %s", $res->getStatusCode(), $res->getBody()));
+            }
+            $body = json_decode($res->getBody(), true);
+            if (!isset($body['access_token'])) {
+                throw new \Exception('Could not retrieve authorization from Partners.');
+            }
+
+            $this->writeTokenCache($body);
+
+            return $body['access_token'];
+        } else {
+            $cachedToken = json_decode(file_get_contents($this->getTokenCacheFilePath()), true);
+
+            return $cachedToken['access_token'];
+        }
+    }
+
+    protected function isCacheExpired()
+    {
+        $stat = @stat($this->getTokenCacheFilePath());
+        if (!$stat) {
+            return true;
+        }
+        $expiresAt = json_decode(file_get_contents($this->getTokenCacheFilePath()), true)['expires_at'];
```

Comment:
you should check than json_decode return is not false. It could happen if content of file is not a json, or if file_get_contents did not success to read the file